### PR TITLE
[prometheus-node-exporter] Update VPA containerPolicies container name to match DaemonSet

### PR DESCRIPTION
**Pull request by hizeph**
#### What this PR does / why we need it

Changes the DaemonSet container name to match the name used by the VerticalPodAutoscaler containerPolicies

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-node-exporter/templates/daemonset.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ spec:
       serviceAccountName: {{ include "prometheus-node-exporter.serviceAccountName" . }}
       containers:
         {{- $servicePort := ternary 8100 .Values.service.port .Values.kubeRBACProxy.enabled }}
-        - name: node-exporter
+        - name: {{ include "prometheus-node-exporter.name" . }}
```
Comment (monotek) [inline charts/prometheus-node-exporter/templates/daemonset.yaml:44]: Imho this makes no sense. 
The vpa does'nt care for the container name.
It looks for " kind: DaemonSet".

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/templates/daemonset.yaml (hunk 1 above)
Comment (monotek) [reply charts/prometheus-node-exporter/templates/daemonset.yaml:44]: I still don't think a dynamic containername is a good idea.

Comment (hizeph) [reply charts/prometheus-node-exporter/templates/daemonset.yaml:44]: That's fair, I'll update the VPA object instead, to match the container name.

**Review by zanhsieh — DISMISSED** [review]

**Comment by zanhsieh** [comment]
@hizeph 
Could you sign DCO plz?

**Comment by hizeph** [comment]
> @hizeph Could you sign DCO plz?

Done, sorry about that

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**